security/src/main/java/io/micronaut/security/endpoints/LogoutController.java

```diff
@@ -42,8 +48,24 @@
 @Secured(SecurityRule.IS_ANONYMOUS)
 public class LogoutController {
 
-    protected final LogoutHandler logoutHandler;
-    protected final ApplicationEventPublisher eventPublisher;
+    private final LogoutHandler logoutHandler;
+    private final ApplicationEventPublisher eventPublisher;
+    private final List<HttpMethod> allowedMethods;
+
+    /**
+     *
+     * @param logoutHandler A collaborator which helps to build HTTP response if user logout.
+     * @param eventPublisher The application event publisher
+     * @param logoutControllerConfiguration Configuration for the Logout controller
+     */
+    @Inject
+    public LogoutController(@Nullable LogoutHandler logoutHandler,
```

Comment:
The old constructor should be deprecated so it can be removed in the next major version